pkg/cli/validate_test.go

```diff
@@ -0,0 +1,265 @@
+	jwks["keys"] = []jwk.Key{ecdsaKey}
+
+	j, err := json.MarshalIndent(jwks, "", " ")
+	if err != nil {
+		t.Fatal("couldn't create jwks json")
+	}
+	path := "/.well-known/jwks"
+	mux := http.NewServeMux()
+	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, "%s", j)
+	})
+
+	svr := httptest.NewServer(mux)
+
+	t.Cleanup(func() {
+		svr.Close()
+	})
+
+	// Build test tokens
+	token := testTokenBuilder(t)
+	if err := jvspb.SetJustifications(token, []*jvspb.Justification{
+		{
+			Category: "explanation",
+			Value:    "test",
+		},
+		{
+			Category: "foo",
+			Value:    "bar",
+		},
+	}); err != nil {
+		t.Fatalf("failed to set justifications in token: %s", err)
+	}
+	signedToken := testSignToken(t, token, privateKey, keyID)
+	breakglassToken, err := jvspb.CreateBreakglassToken(token, "prod is down")
+	if err != nil {
+		t.Fatalf("failed to build breakglass token: %s", err)
+	}
+
+	cases := []struct {
+		name   string
+		config *config.CLIConfig
+		args   []string
+		pipe   bool
+		expOut string
+		expErr string
+	}{
+		{
+			name:   "too_many_args",
+			args:   []string{"foo"},
+			expErr: `accepts 0 arg(s)`,
+		},
+		{
+			name:   "missing_token",
+			args:   nil,
+			expErr: `"token" not set`,
+		},
+		{
+			name: "invalid_token",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args:   []string{"-t=invalid token"},
+			expErr: `failed to parse token headers`,
+		},
+		{
+			name: "signed",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", signedToken},
+			expOut: `-------RESULT-------
+validated!
+breakglass  false
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "breakglass",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", breakglassToken},
+			expOut: `-------RESULT-------
+validated!
+breakglass  true
+----JUSTIFICATION----
+breakglass  "prod is down"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "signed_from_pipe",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", "-"},
+			pipe: true,
+			expOut: `-------RESULT-------
+validated!
+breakglass  false
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cmd := newValidateCmd(tc.config)
+			var stdout string
+			var gotErr error
+			if tc.pipe {
```

Comment:
I think we're over-complicating things here.

Reply:
Any suggestions?

Reply:
For the `io.Reader`, you can use https://pkg.go.dev/strings#Builder instead of writing to a file

Reply:
accidently found out that strings.NewReader(signedToken) is working as well?